src/hooks/useAddLiquidityHandler.tsx

```diff
@@ -104,6 +107,32 @@ export const useAddLiquidityHandler = () => {
     fetchContract()
   }, [provider])
 
+  useEffect(() => {
+    const getTokenABalances = async () => {
+      if (provider) {
+        let accountAddress = accountInfo?.address || ''
+        if (tokenASelected.address) {
+          let balance = await getTokenBalance(accountAddress, tokenASelected.address, tokenASelected.symbol, provider)
+          setTokenABalance(balance || '0')
+        }
+      }
+    }
+    getTokenABalances()
+  }, [tokenASelected, provider])
+
+  useEffect(() => {
+    const getTokenBBalances = async () => {
```

Comment:
You are only getting one balance. Rename to getTokenBalanceAsync and move it out of the useEffect to be called by the useEffect so it won't be duplicate code. Actually even better is to create the getTokenBalanceAsync  in the getSmartContract.tsx.

Reply:
But I will still have to wait for the data to be returned. So I will again have to make an async method inside of use effect